soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -120,6 +129,18 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 	nrf_regulators_dcdcen_vddh_set(NRF_REGULATORS, true);
 #endif
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
+	DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder)
+	static const uint8_t pf_psels[] = {
```

Comment:
Remove static. I don't see the need to retain this state between function calls.

Reply:
`static const` is intentional, please have a look on https://github.com/zephyrproject-rtos/zephyr/pull/42576#discussion_r802754077 and following comment https://github.com/zephyrproject-rtos/zephyr/pull/42576#discussion_r802825648

Reply:
Interesting.

So with static it reads the array from flash, but without static it writes the values from flash to the stack, and then reads from the stack.

TIL, thanks.